### ISSUE TITLE
[11.0][web_widget_x2many_2d_matrix] Problem with limit on matrix

### ISSUE THREAD

**Issue by agranadosb** [OCA/web#1012]
The limit attribute makes that the matrix does not have more than 40 element, because default limit is 40.
Problems that appear when rows x columns > limit = 40:
1.- Undefined col
2.- If the number of columns of a row is divider of the limit, the matrix only shows the rows that have (in total) 40 elements
Steps to example:
Matrix with 5 columns and 9 or more rows, the widget only show the first 8 rows. 8*5=40, the default limit of One2many
If we have 6 columns and 7 or more rows, odoo will throw an error, because 6 dont is divider of 40.

For the moment, my solution is change the limit to Infinty on:
init_matrix: function(){
      var self = this,
          records = self.recordData[this.name].data;
      // Wipe the content if something still exists
      this.by_x_axis = {};
      this.by_y_axis = {};

      this.attrs.limit = Infinity; <---------------

Regards.

**Comment by pedrobaeza** [comment]
This was solved manually on hr_timesheet_sheet this way:

https://github.com/OCA/hr-timesheet/pull/128

@jbeficent @simahawk what do you think about this change?

**Comment by agranadosb** [comment]
Thanks @pedrobaeza ,
i didnt know that issue was open.
Regrads.

**Comment by simahawk** [comment]
the workaround seems good. I'd keep the comment in the fix and update the ROADMAP to keep track.

**Comment by pedrobaeza** [comment]
Can then any of you make the PR?

**Comment by agranadosb** [comment]
I made the pull request.
Regards.

**Comment by tonihr** [comment]
Hi,
I'm following this Issue and  PR, and the problem is derived from AbstractView of odoo in web module.

I have made the following implementation and it has worked correctly, which is your opinion:

```
odoo.define('web_widget_x2many_2d_matrix.matrix_limit_extends', function (require) {
    "use strict";

var AbstractView = require('web.AbstractView');

var AbstractViewCustom = AbstractView.include({

    _setSubViewLimit: function (attrs) {
	this._super(attrs);

        var view = attrs.views && attrs.views[attrs.mode];
        var limit = view && view.arch.attrs.limit && parseInt(view.arch.attrs.limit);
	if (!limit && attrs.widget === 'x2many_2d_matrix') {
            limit = Infinity;
	    attrs.limit = limit || 40;
        }
        
    }

});

return AbstractViewCustom;

});
```

Regards,

**Comment by JordiBForgeFlow** [comment]
@tonihr Your proposal works like a charm. I have implemented it here: https://github.com/Eficent/web/tree/11_fix_matrix_limit

Will you send a PR or I sumbit mine?

**Comment by tonihr** [comment]
@jbeficent If have it implemented and works fine, submit.

**Comment by pedrobaeza** [comment]
Already fixed in #1014
